src/main/java/apoc/load/LoadXls.java

```diff
@@ -172,17 +180,60 @@ private Object convertArray(Object value) {
 
         private Object convertType(Object value) {
             if (nullValues.contains(value) || value == null) return null;
-//            if (!(value instanceof String)) return value;
             switch (type) {
-                case STRING: return value.toString();
-                case INTEGER: return Util.toLong(value);
-                case FLOAT: return Util.toDouble(value);
-                case BOOLEAN: return Util.toBoolean(value);
-                case NULL: return null;
-                case LIST: return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
-                default: return value;
+                case STRING:
+                    if (value instanceof Temporal && dateFormat instanceof String && dateFormat != null) {
+                        return DateTimeFormatter.ofPattern(dateFormat.toString()).format((TemporalAccessor) value);
+                    } else {
+                        return value.toString();
+                    }
+                case INTEGER:
+                    return Util.toLong(value);
+                case FLOAT:
+                    return Util.toDouble(value);
+                case BOOLEAN:
+                    return Util.toBoolean(value);
+                case NULL:
+                    return null;
+                case LIST:
+                    return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
+                case DATE:
+                    return temporalFormat(value, LocalDate.class);
+                case DATE_TIME: case LOCAL_DATE_TIME:
+                    return temporalFormat(value, LocalDateTime.class);
+                case LOCAL_TIME: case TIME:
+                    return temporalFormat(value, LocalTime.class);
+                default:
+                    return value;
             }
         }
+
+        private Object temporalFormat(Object value, Class<?> temporalClass) {
+            if (value instanceof String) {
+                try {
+                    switch (temporalClass.getSimpleName()){
+                        case "LocalDate":
+                            return LocalDate.parse(value.toString(), getFormat(dateFormat));
+                        case "LocalDateTime":
+                            return LocalDateTime.parse(value.toString(), getFormat(dateFormat));
+                        case "LocalTime":
+                            return LocalTime.parse(value.toString(), getFormat(dateFormat));
+                    }
+                } catch (DateTimeException e) {
+                    return value;
+                }
+            }
+            return value;
+        }
+
+        private DateTimeFormatter getFormat(Object value) {
+                if (value instanceof List) {
+                    return DateTimeFormatter.ofPattern((String) ((ArrayList) dateFormat).stream().map(item -> "[" + item + "]").collect(Collectors.joining("")));
```

Comment:
@AngeloBusato questo non va bene se non riconosce un pattern ritorna un eccezione, invece deve continuare finche non ne trova uno buono:
https://repl.it/@conker84/DateTimeFormatter-Parse?language=java